src/frontend/src/optimizer/plan_node/stream_table_scan.rs

```diff
@@ -192,6 +192,37 @@ impl StreamTableScan {
             .build(vec![0], 1)
             .with_id(state.gen_table_id_wrapped())
     }
+
+    /// Build catalog for cdc backfill state
+    /// Right now we only persist whether the backfill is finished and the corresponding cdc offset
+    /// schema: | `table_id` | `backfill_finished` | `row_count` | `cdc_offset` |
+    pub fn build_cdc_backfill_state_catalog(
+        &self,
+        state: &mut BuildFragmentGraphState,
+    ) -> TableCatalog {
+        let properties = self.ctx().with_options().internal_table_subset();
+        let mut catalog_builder = TableCatalogBuilder::new(properties);
+
+        // use `table_id` as primary key in state table.
+        catalog_builder.add_column(&Field::with_name(DataType::Varchar, "split_id"));
+        catalog_builder.add_order_column(0, OrderType::ascending());
+
+        catalog_builder.add_column(&Field::with_name(
+            DataType::Boolean,
+            format!("{}_backfill_finished", self.table_name()),
```

Comment:
I suggest we don't put the external `table_name` here as the prefix of the column name so that all table's backfill state share the same schema. It is easy to construt debugging SQL query in this case.

Reply:
agree, fixed.